### PR TITLE
Allow conditional skip with skip_snapshot_verify

### PR DESCRIPTION
**Pull request by dominikschubert**
Example usage: `@pytest.mark.skip_snapshot_verify(condition=lambda: 1 == 1)`

The condition always needs to be a callable that will be evaluated to determine if the verification should be skipped or not.

This PR also makes sure that verification of snapshots is never skipped when running with `TEST_TARGET=AWS_CLOUD`.

### CODE REVIEW

**Review by steffyP — APPROVED** [review]
LGTM! 👍 Thanks for adding this feature 🙂

File: localstack/testing/pytest/snapshot.py [hunk 1]
```diff
@@ -81,6 +104,7 @@ def fixture_region():
 
 
 @pytest.fixture(name="snapshot", scope="function")
+@pytest.mark.skipif(condition="")
```
Comment (dfangl) [inline localstack/testing/pytest/snapshot.py:107]: wat

**Review by dfangl — APPROVED** [review]
LGTM, just a minor nit

File: localstack/testing/pytest/snapshot.py (hunk 1 above)
Comment (dfangl) [inline localstack/testing/pytest/snapshot.py:107]:
```suggestion
```

**Pull request merged**